### DISCUSSION TITLE
Getting "AuthSessionMissingError: Auth session missing!" in Next.js 14 when getUser()

### DISCUSSION THREAD

**Discussion by timoteioros** [supabase/supabase#27492] [Questions] [answered]
Hi there. First time trying both next.js 14 and supabase. I am facing an issue with some auth stuff. I have a server component Navbar like this:

`
...
const Navbar = async () => {
  const { user, error } = await getUserAction();
  console.log(error);

  const { avatar_url } = user?.user_metadata || {};

  return ( ... );
};
...
`

With this getUserAction:

`const getUserAction = async () => {
  const supabase = createSupabaseServerClient();

  const {
    data: { user },
    error,
  } = await supabase.auth.getUser();

  return { user, error };
};`

When I go to the app in the browser I get in the server console the error in the title. Yes, I have no signed in user, that's right, but why do I get this error? I would expect to just get a null user. Also when trying to SignIn with google OAuth nothing happens, it just refreshes but no signed in user, might be still because of the error above. Won't paste the code for SignIn here as well, but I will put the github repo with the full code (there's not much code yet):

https://github.com/timoteioros/salariimedici

**Comment by GaryAustin1** [comment]
getUser errors if there is no session in the supabase client or you don't provide.  getSession will return null if there is no session.

**Reply by timoteioros** [reply]
Being new to all things supabase, how might I give it a session? I did follow the docs guides of next.js though, and I don't recall there being anything memorable about session. Thanks!

**Reply by timoteioros** [reply, marked as answer]
Ok so I now I get it. My problem was that I was trying to oauth on the server, but I didn't redirect, as per the docs:

```
const { data, error } = await supabase.auth.signInWithOAuth({
  provider,
  options: {
    redirectTo: 'http://example.com/auth/callback',
  },
})

if (data.url) {
  redirect(data.url) // use the redirect API for your server framework
}
```

And now I understand that I will always get that error when I have no user signed in.
